magma/bit.py

```diff
@@ -161,7 +161,13 @@ def ite(self, t_branch, f_branch):
         # Note: coreir flips t/f cases
         # self._mux monkey patched in magma/primitives/mux.py to avoid circular
         # dependency
-        return self._mux([f_branch, t_branch], self)
+        r = self._mux([f_branch, t_branch], self)
+
+        # cast protocol values back to protocol
```

Comment:
wondering if we want to add the conversion function in `Mux` generator (https://github.com/phanrahan/magma/blob/0975d81d192918dcee16194e03b4f1912c547841/magma/primitives/mux.py#L38) or `mux` helper function (https://github.com/phanrahan/magma/blob/0975d81d192918dcee16194e03b4f1912c547841/magma/primitives/mux.py#L126).

@leonardt

Reply:
doesn't matter to me I was just was looking the fastest way to get a working solution

Reply:
ok give me a little bit, i'll fix the mux and it should fix your issue

Reply:
Thanks, sounds good to me.  As long as it passes my test I don't care where the casting logic is.